packages/core/src/codewhisperer/commands/invokeRecommendation.ts

```diff
@@ -52,5 +55,9 @@ export async function invokeRecommendation(
         )
     }
 
-    await RecommendationService.instance.generateRecommendation(client, editor, 'OnDemand', config, undefined)
+    const traceId = randomUUID()
+    TelemetryHelper.instance.setTraceId(traceId)
```

Comment:
Do race conditions matter?
Can you have code that is running and should be assigned to trace id #1, but auto/manual trigger occurs and overwrites the current traceId in a new context while other code is still executing?

Reply:
I don't think so, the current codewhisperer https://github.com/aws/aws-toolkit-vscode/blob/master/packages/core/src/codewhisperer/util/telemetryHelper.ts does the same thing. It stores the states until for a full invocation session with only one session active at a time